Userland/Libraries/LibWeb/CSS/SelectorEngine.cpp

```diff
@@ -78,23 +78,56 @@ static inline bool matches_checked_pseudo_class(DOM::Element const& element)
 
 static inline bool matches_attribute(CSS::Selector::SimpleSelector::Attribute const& attribute, DOM::Element const& element)
 {
-    switch (attribute.match_type) {
-    case CSS::Selector::SimpleSelector::Attribute::MatchType::HasAttribute:
+    if (attribute.match_type == CSS::Selector::SimpleSelector::Attribute::MatchType::HasAttribute) {
+        // early way out in case of an attribute existence selector
```

Comment:
Please make sure comments follow the [coding style](https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#comments)